unitex/src/fr/umlv/unitex/config/Config.java

```diff
@@ -1585,4 +1614,133 @@ public static SvnMonitor getSvnMonitor() {
         }
         return monitor;
     }
+    
+    public static boolean isRunning(final String[] args) {
+    	try {
+			// Issue #27 use ServerSocket to restrict IDE to single instance and to allow failed instance to pass file name to running instance.
+			// Run in SwingWorker background thread so doesn't block main thread
+			serverSocket = new ServerSocket(PORT,0,InetAddress.getByAddress(new byte[] {127,0,0,1}));  
+			SOCKET_OPEN = true;
+			
+			SwingWorker<String, Void> socketThread = new SwingWorker<String, Void>() { 
+				
+	            @Override
+	            public String doInBackground() { 
+	            	while (SOCKET_OPEN) {
+	            		serverSocketListener();
+	            	}
+	                return "";
+	            }	
+	        };
+	        socketThread.execute(); 
+			return false;
+		} catch (BindException e) {
+			System.err.println("IDE instance already running.");
+			clientSocketListener(args);
+			return true;
+	    } catch (IOException e) {
+	        e.printStackTrace();
+	        return true;
+	    }
+    }
+    
+    /**
+     * Extracts the graph files from args passed to application on startup and opens graph frames
+     * Issue #27
+     */
+    public static void openGraphFiles(String[] args) {
+    	for (int i = 0; i < args.length; i++) {
+    		if (args[i].indexOf("=") == -1) {
+    			File f = new File(args[i]);
```

Comment:
I've fixed this to check for the grf extension. However are you saying that pre 3.1 the UnitexToolLogger path can be passed on the command line without -Dunitex.binary.dir=